src/include/btree_cmp_inline.h

```diff
@@ -31,70 +207,30 @@
 static WT_INLINE int
 __wt_lex_compare(const WT_ITEM *user_item, const WT_ITEM *tree_item)
 {
-    size_t len, usz, tsz;
-    const uint8_t *userp, *treep;
+    size_t len, tsz, usz;
+    int lencmp, ret_val;
 
     usz = user_item->size;
     tsz = tree_item->size;
-    len = WT_MIN(usz, tsz);
-
-    userp = (const uint8_t *)user_item->data;
-    treep = (const uint8_t *)tree_item->data;
-
-#ifdef HAVE_X86INTRIN_H
-    /* Use vector instructions if we'll execute at least 2 of them. */
-    if (len >= WT_VECTOR_SIZE * 2) {
-        size_t remain;
-        __m128i res_eq, u, t;
-
-        remain = len % WT_VECTOR_SIZE;
-        len -= remain;
-        if (WT_ALIGNED_16(userp) && WT_ALIGNED_16(treep))
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_load_si128((const __m128i *)userp);
-                t = _mm_load_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        else
-            for (; len > 0;
-                 len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-                u = _mm_loadu_si128((const __m128i *)userp);
-                t = _mm_loadu_si128((const __m128i *)treep);
-                res_eq = _mm_cmpeq_epi8(u, t);
-                if (_mm_movemask_epi8(res_eq) != 65535)
-                    break;
-            }
-        len += remain;
-    }
-#elif defined(HAVE_ARM_NEON_INTRIN_H)
-    /* Use vector instructions if we'll execute at least 1 of them. */
-    if (len >= WT_VECTOR_SIZE) {
-        size_t remain;
-        uint8x16_t res_eq, u, t;
-        remain = len % WT_VECTOR_SIZE;
-        len -= remain;
-        for (; len > 0; len -= WT_VECTOR_SIZE, userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
-            u = vld1q_u8(userp);
-            t = vld1q_u8(treep);
-            res_eq = vceqq_u8(u, t);
-            if (vminvq_u8(res_eq) != 255)
-                break;
-        }
-        len += remain;
+    if (usz < tsz) {
+        len = usz;
+        lencmp = -1;
+    } else if (usz > tsz) {
+        len = tsz;
+        lencmp = 1;
+    } else {
+        len = usz;
+        lencmp = 0;
```

Comment:
```
   0x00000000000c1588 <+44>:	cmp	x10, x9
   0x00000000000c158c <+48>:	csel	x9, x9, x10, hi  // hi = pmore
   0x00000000000c1590 <+52>:	cset	w11, hi  // hi = pmore
   0x00000000000c1594 <+56>:	csel	x2, x10, x9, cc  // cc = lo, ul, last
```
Here's what I got. Looks very similar.

Reply:
Thanks for double checking! I just wanted to ensure it didn't end up with `b.XX` conditional branches in there. Clang is very aggressive and uses conditional selections pretty much whenever it can. GCC is much more finicky and I still don't have a great intuition for when it will or won't without massaging and looking at the generated ASM.